articles/quickstart.md

```diff
@@ -1,163 +1,129 @@
-        Message("Hello quantum world!");
-    }
-}
-```
+### About Q# operations
 
-Now, replace `Message("Hello quantum world!");` with the following code between the braces that enclose the operation body:
+An operation is the basic unit of quantum execution in Q#. It is roughly equivalent to a function in C or C++ or Python, or a static method in C# or Java.
 
-```qsharp
-        if (desired != M(q1)) {
-            X(q1);
-        }
-```
+The arguments to an operation are specified as a tuple, within parentheses.
 
-The file should now look like:
+The return type of the operation is specified after a colon. In this case, the `Set` operation has no return, so it is marked as returning `Unit`. This is the Q# equivalent of `unit` in F#, which is roughly analogous to `void` in C#.
 
-```qsharp
-namespace Quantum.Bell {
-    open Microsoft.Quantum.Intrinsic;
-    open Microsoft.Quantum.Canon;
+### About quantum gates
 
-    operation Set (desired: Result, q1: Qubit) : Unit {
-        if (desired != M(q1)) {
-            X(q1);
-        }
-    }
-}
-```
+You have used two quantum gates in your first Q# operation:
+
+* The [M](xref:microsoft.quantum.intrinsic.m) gate, which measures the state of the qubit
+* The [X](xref:microsoft.quantum.intrinsic.x) gate, which flips the state of a qubit
+
+A quantum gate transforms the state of a qubit. The name comes from the analogy of a classical logic gate such as a `NOT` gate, which inverts the state of a classical bit.
+
+### Add Q# test code
 
-#### Set or flip a qubit
-This operation may now be called to set a qubit in a known state (`Zero` or `One`). We measure the qubit, if it's in the state we want, we leave it alone, otherwise, we flip it with the `X` gate.
-
-> [!NOTE]
-> This code defines a Q# __operation__.
-> An operation is the basic unit of quantum execution in Q#.
-> It is roughly equivalent to a function in C or C++ or Python,
-> or a static method in C# or Java.
->
-> The arguments to an operation are specified as a tuple, within parentheses.
-> The return type of the operation is specified after a colon.
-> In this case, the `Set` operation has no return, so it is marked as
-> returning `Unit`.
-> This is the Q# equivalent of `unit` in F#, which is roughly analogous to
-> `void` in C#.
-> 
-> An operation has a __body__ section, which contains the implementation
-> of the operation.
-> An operation may also have __adjoint__, __controlled__, and 
-> __controlled adjoint__ sections.
-> These are used to specify specific variants of appropriate operations.
-> See the [Q# language reference](xref:microsoft.quantum.language.intro) for more
-> information.
-
-#### Measure a qubit
-Add the following operation to the namespace, after the end of the
+1. Add the following operation to the `Bell.qs` file, inside the namespace, after the end of the
 `Set` operation:
 
-```qsharp
+    ```qsharp
     operation BellTest (count : Int, initial: Result) : (Int, Int) {
```

Comment:
I'd suggest that this is a good opportunity to rename operations and functions in the documentation to better align with the style guide. For instance, in the original quickstart, the name `BellTest` isn't a verb, such that it's not clear that it has a side effect on the quantum device. Especially since there's a loosely related other concept in quantum computing known as a "Bell test," (see the CHSH game sample), perhaps we could rename to `TestBellStatePreparation` or something similar?